pkg/cloudprovider/provider/baremetal/plugins/tinkerbell/client/template.go

```diff
@@ -20,73 +20,252 @@ import (
 	"context"
 	"fmt"
 
-	"github.com/tinkerbell/tink/protos/template"
+	tinkv1alpha1 "github.com/tinkerbell/tink/api/v1alpha1"
+	"gopkg.in/yaml.v3"
+
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
-// Template client for Tinkerbell.
+type Task struct {
+	Name        string            `json:"name"`
+	WorkerAddr  string            `json:"worker" yaml:"worker"`
+	Actions     []Action          `json:"actions"`
+	Volumes     []string          `json:"volumes,omitempty"`
+	Environment map[string]string `json:"environment,omitempty"`
+}
+
+// Action represents a workflow action.
+type Action struct {
+	Name        string            `json:"name,omitempty"`
+	Image       string            `json:"image,omitempty"`
+	Timeout     int64             `json:"timeout,omitempty"`
+	Volumes     []string          `json:"volumes,omitempty"`
+	Pid         string            `json:"pid,omitempty"`
+	Environment map[string]string `json:"environment,omitempty"`
+	Command     []string          `json:"command,omitempty"`
+}
 type Template struct {
-	client template.TemplateServiceClient
+	Version       string `yaml:"version"`
+	Name          string `yaml:"name"`
+	GlobalTimeout int64  `yaml:"global_timeout"`
+	Tasks         []Task `yaml:"tasks"`
 }
 
-// NewTemplateClient returns a Template client.
-func NewTemplateClient(client template.TemplateServiceClient) *Template {
-	return &Template{client: client}
+const (
+	fsType             = "ext4"
+	defaultInterpreter = "/bin/sh -c"
+	hardwareDisk1      = "{{ index .Hardware.Disks 0 }}"
+
+	ProvisionWorkerNodeTemplate = "provision-worker-node"
+)
+
+// TemplateClient handles interactions with the Tinkerbell Templates in the Tinkerbell cluster.
+type TemplateClient struct {
+	tinkclient client.Client
 }
 
-// Get returns a Tinkerbell Template.
-func (t *Template) Get(ctx context.Context, id, name string) (*template.WorkflowTemplate, error) {
-	req := &template.GetRequest{}
-	if id != "" {
-		req.GetBy = &template.GetRequest_Id{Id: id}
-	} else {
-		req.GetBy = &template.GetRequest_Name{Name: name}
+// NewTemplateClient creates a new client for managing Tinkerbell Templates.
+func NewTemplateClient(k8sClient client.Client) *TemplateClient {
+	return &TemplateClient{
+		tinkclient: k8sClient,
 	}
+}
 
-	tinkTemplate, err := t.client.GetTemplate(ctx, req)
-	if err != nil {
-		if err.Error() == sqlErrorString || err.Error() == sqlErrorStringAlt {
-			return nil, fmt.Errorf("template %w", ErrNotFound)
-		}
-
-		return nil, fmt.Errorf("getting template from Tinkerbell: %w", err)
+func (t *TemplateClient) Delete(ctx context.Context, namespacedName types.NamespacedName) error {
+	template := &tinkv1alpha1.Template{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      namespacedName.Name,
+			Namespace: namespacedName.Namespace,
+		},
+	}
+	// Create the Template object in the Tinkerbell cluster
```

Comment:
This comment is wrong.

Reply:
This is part of the leftovers. Will be removed.